docs/user-guide/gateway.md

```diff
@@ -0,0 +1,221 @@
+Each peering can be configured to provide the necessary services for traffic that uses that peering.
+
+### Simple Gateway Peering Between VPCs
+
+```{.yaml .annotate linenums="1" title="gw-peer.yaml"}
+apiVersion: gateway.githedgehog.com/v1alpha1
+kind: Peering
+metadata:
+  name: vpc-1--vpc-2
+  namespace: default
+spec:
+  peering:
+    vpc-1:
+      expose:
+        - ips:
+          - cidr: 10.0.0.0/24 # Expose all IP address in the 10.0.0.0/24 CIDR block to vpc-2
+    vpc-2:
+      expose:
+        - ips:
+          - cidr: 192.168.0.0/16 # Expose all IP addresses in the 192.168.0.0/16 CIDR block to vpc-1
+```
+
+A simple peering with no services deployed between the VPCs. This traffic will
+transit the gateway node(s).
+
+### Gateway Peering with stateless NAT
+
+```{.yaml .annotate linenums="1" title="sl-gw-peer.yaml"}
+apiVersion: gateway.githedgehog.com/v1alpha1
+kind: Peering
+metadata:
+  name: vpc-1--sl-nat--vpc-2
+  namespace: default
+spec:
+  peering:
+    vpc-1:
+      expose:
+        - ips:
+          - cidr: 10.0.1.0/24 # Expose all IP address in the 10.0.1.0/24 CIDR block to vpc-2
+          as:
+          - cidr: 10.11.11.0/24
+        - ips:
+          - cidr: 10.0.2.3/32 # Expose the single IP Statless NAT mapping to vpc-2
+          as:
+          - cidr: 10.11.22.3/32
+    vpc-2:
+      expose:
+        - ips:
+          - cidr: 10.0.2.0/24
+          as:
+          - cidr: 10.22.22.0/25
+          - cidr: 10.22.22.128/25
+
+
+```
+
+Stateless NAT translates source and destination IP addresses for all packets that traverse
+the peering, but it does not maintain any flow state for the connection. A 
+one-to-one mapping is established between the addresses exposed in the CIDRs for
+`ips` and the addresses to use represented by the CIDRs in
+`as:` each address from the first group is consistently mapped to a single
+address from the second group. Therefore, the total number of addresses covered
+by the CIDRs YAML array entries from `ips` must be equal to the total number of
+addresses covered by the CIDRs from `as`.
+
+### Gateway Peering with Stateful Source NAT
+
+```{.yaml .annotate linenums="1" title="gw-sf-nat-peer.yaml"}
+apiVersion: gateway.githedgehog.com/v1alpha1
+kind: Peering
+metadata:
+  name: vpc-1--sf-nat--vpc-2
+  namespace: default
+spec:
+  peering:
+    vpc-1:
+      expose:
+        # Allow 10.0.0.0/24 addresses to talk to vpc-2
+        # Because of stateful NAT, traffic from vpc-2 to vpc-1 is only allowed if there is
+        # a flow table entry created by a traffic initiated from vpc-1 to vpc-2.
+        - ips:
+          - cidr: 10.0.0.0/24
+          as:
+          - cidr: 10.0.1.0/31  # but, NAT those addresses using the addresses in 10.0.1.0/31
+          nat:  # Contains the nat configuration
+            stateful:  # Make NAT stateful, connections initiated from vpc-1 to vpc-2 will be added to the flow table
```

Comment:
```suggestion
            stateful:  # Make NAT stateful, connections initiated from vpc-1 to vpc-2 will be added to the flow table.
                       # An entry for the reverse direction will be added too, so that the receiving endpoint in vpc-2 can reply.
```